### PR TITLE
[full-ci] Creating user via Graph in e2e tests and UI tests

### LINKED ISSUE #6836: make e2e tests work with GraphAPI of ocis
similar to https://github.com/owncloud/owncloud-test-middleware/pull/107 & https://github.com/owncloud/ocis/pull/3404

### PR DESCRIPTION
**Pull request by ScharfViktor**
Step 1:
- e2e tests with ocis use graphApi

Steps 2: 
- UI tests with ocis use graphApi (changes in middleware)

Fixes #6836

### CODE REVIEW

**Review by pascalwengerter — CHANGES_REQUESTED** [review]
Missing the middleware version bump in .drone.star

**Comment by pascalwengerter** [comment]
> Missing the middleware version bump in .drone.star

I'm an idiot, we bumped it already yesterday as part of the Uppy-PR. Happy to merge if CI turns green

**Comment by ScharfViktor** [comment]
> Missing the middleware version bump in .drone.star


https://github.com/owncloud/web/blob/97097063e15d2fe7961bcec863afaa540e17375e/.drone.star#L23

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingAutocompletion</strong> https://drone.owncloud.com/owncloud/web/24974/63/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingAutocompletion-shareAutocompletion_feature-L182.png
  

### webUISharingAutocompletion-shareAutocompletion_feature-L56.png
  

### webUISharingAutocompletion-shareAutocompletion_feature-L66.png
  


</p></details>

File: tests/e2e/support/api/graph/index.ts
Comment (fschade) [inline tests/e2e/support/api/graph/index.ts:2]: can be handled in one export e.g. `export { createUser, deleteUser } from './user'` or even `export * from './user'`

File: tests/e2e/support/api/graph/user.ts
Comment (fschade) [inline tests/e2e/support/api/graph/user.ts:13]: `const response = await request({
    method: 'POST',
    path: join('graph', 'v1.0', 'users'),
    body,
    user: admin
  })`

File: tests/e2e/support/api/graph/user.ts
Comment (fschade) [inline tests/e2e/support/api/graph/user.ts:5]: can we return the user here too? `: Promise<User>`

**Review by fschade — CHANGES_REQUESTED** [review]
found some smaller improvements, nothing crucial.
The me.ts file looks obsolete to me, can we refactor the ma func into user.ts too?

**Comment by ScharfViktor** [comment]
> 

Good idea. I'll move "me" to "user"

**Comment by ScharfViktor** [comment]
e2e tests passed. UI tests as expected failed. I'm doing fix after review and go to step 2

**Comment by ScharfViktor** [comment]
CI is green. could you do a review?

before merge: need change - OCIS_COMMITID and  OC_TESTING_MIDDLEWARE

**Review by pascalwengerter — APPROVED** [review]
LGTM, please use the squash-merge option :) we need to take care of the flaky test next!

**Review by fschade — APPROVED** [review]
LGTM 🎸

**Pull request merged**
